go/ql/lib/semmle/go/Types.qll

```diff
@@ -997,6 +1014,17 @@ class NamedType extends @namedtype, CompositeType {
   }
 
   override Type getUnderlyingType() { result = this.getBaseType().getUnderlyingType() }
+
+  override predicate hasLocationInfo(
+    string filepath, int startline, int startcolumn, int endline, int endcolumn
+  ) {
+    exists(DeclaredType dt |
+      dt.getType() = this and
+      // Note that if the type declaration isn't in the source that we have
+      // then we use a dummy location.
+      dt.hasLocationInfo(filepath, startline, startcolumn, endline, endcolumn)
```

Comment:
I think we need to explicitly return the dummy location when there isn't such a DeclaredType